python/ray/data/dataset.py

```diff
@@ -1336,10 +1337,14 @@ def filter(
     def repartition(
         self,
         num_blocks: int,
+        max_num_rows_per_block: Optional[int] = None,
         *,
         shuffle: bool = False,
     ) -> "Dataset":
-        """Repartition the :class:`Dataset` into exactly this number of :ref:`blocks <dataset_concept>`.
+        """Repartition the :class:`Dataset` into exactly this number of
+        :ref:`blocks <dataset_concept>`. When `max_num_rows_per_block` is set, it
+        overrides `num_blocks` and repartitions :class:`Dataset` to honor at most
+        maximum rows per :ref:`blocks <dataset_concept>`.
```

Comment:
Nit: For user-facing docstrings, can we adhere to the convention of one-line summaries